src/utilities.jl

```diff
@@ -0,0 +1,89 @@
+using MCMCChains, StatsFuns
+export pointwise_loglikes, compute_loo
+
+"""
+    pointwise_loglikes(chain::Chains, data, ll_fun)
+
+Computes the pointwise log likelihoods where [d,s,c] corresponds to log likelihood of 
+evaluated at datapoint d, sample s, for chain c. 
+
+- `chain`: a chain object from MCMCChains
+- `data`: a vector of data used to estimate parameters of the model 
+- `ll_fun`: a function that computes the log likelihood of a single data point: f(θ1, ..., θn, data), where θi is the ith parameter
+"""
+function pointwise_loglikes(chain::Chains, data, ll_fun)
+    samples = Array(Chains(chain, :parameters).value)
+    pointwise_loglikes(samples, data, ll_fun)
+end
+
+"""
+    pointwise_loglikes(samples::Array{Float64,3}, data, ll_fun)
+
+Computes the pointwise log likelihood 
+
+- `samples`: a three dimensional array of mcmc samples where the first dimension represents the samples, the second dimension represents
+the samples, and the third dimension represents the chains. 
+- `data`: a vector of data used to estimate parameters of the model 
+- `ll_fun`: a function that computes the log likelihood of a single data point: f(θ1, ..., θn, data), where θi is the ith parameter
+"""
+function pointwise_loglikes(samples::Array{Float64,3}, data, ll_fun)
+    fun = (p,d)-> ll_fun(p..., d)
+    n_data = length(data)
+    n_samples, n_chains = size(samples)[[1,3]]
```

Comment:
`n_iterations, n_data_points, n_chains = size(samples)` is cleaner and easier to understand.

Reply:
The dimensions for `samples`  is (sample, parameter, chain). 

But I agree there is a better way to do this:

  `  n_samples, _ ,n_chains = size(samples)`